build_artifacts/v1/v1.0/v1.0.0/v1.0.0-beta/cpu.env.in

```diff
@@ -21,6 +21,13 @@ conda-forge::sagemaker-python-sdk[version='>=2.173.0,']
 conda-forge::supervisor[version='>=4,']
 conda-forge::autogluon[version='>=0.8.2,']
 conda-forge::jupyter-ai[version='>=2.2.0,']
+conda-forge::jupyter-scheduler[version='>=2.2.0,']
+#####################################################################
+# jupyter_core 5.3.2+ breaks compatibility w/ scheduler.
+# This block can be removed once the bug is fixed.
+# See: https://github.com/jupyter-server/jupyter-scheduler/issues/435
+conda-forge::jupyter_core[version='>=5,<=5.3.1']
```

Comment:
Rather than managing/pinning versions of downstream dependencies like this in sagemaker-distribution, ideally jupyter-scheduler itself should identify the jupyter_core constraint in it's own recipe, and the sagemaker-distribution build will respect that when the environment is resolved. Does it?

Reply:
@just4brown The guidance I received was that any new packages that need to be added to the image must be added before the 1.0.0 release, which is currently scheduled for next Monday (10/23/2023) according to my source. I'm not sure if I can get a new version of `jupyter-scheduler` released by then to be included in this image (and allow us to drop these changes to the `*.env.in` files).

This bug is only happening in specific environments like JupyterHub or this image, which is why this is a bug in the latest release.

Reply:
Specifying an upperbound for the jupyter_core dependency could complicate future releases, so I'd like to make sure that is followed up. Can you create an issue for the jupyter-scheduler feedstock and provide an ecd for when the recipe will be updated?

Reply:
Well, this is an issue with the underlying PyPI package as well, so it makes more sense to use the original issue here: https://github.com/jupyter-server/jupyter-scheduler/issues/435

Once I get this bug fixed, I will cut a release on PyPI and Conda Forge (if necessary), then proceed to update any lower bounds in this image and remove the version ceiling I imposed. I plan on having this done by 10/30/2023. I've added it to our team's internal roadmap, DM me if you wish to verify.

Reply:
Works for me, thanks!